### PR TITLE
Add functional tests

### PR DESCRIPTION
**Pull request by MatkovIvan**
* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests if this modifies the Windows code?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Add more functional tests

## Related PRs or issues

[AB#64396](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/64396)
#1282, #1287, #1290, #1291

### CODE REVIEW

**Comment by MatkovIvan** [comment]
The fixes from #1298 should be applied first

File: SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Android/Crashes.cs [hunk 1]
```diff
@@ -295,5 +295,10 @@ public bool ShouldProcess(AndroidErrorReport androidReport)
                 return ShouldProcessErrorReport(report);
             }
         }
+
+        internal static void PlatformUnsetInstance()
```
Comment (guperrot) [inline SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Android/Crashes.cs]:
```suggestion
        private static void PlatformUnsetInstance()
```

File: SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Windows.Shared/Crashes.cs [hunk 2]
```diff
@@ -482,5 +482,10 @@ private ErrorReport MapLogEventToReportAndDeleteOnLastEvent(ChannelEventArgs cha
             }
             return null;
         }
+
+        internal static void PlatformUnsetInstance()
```
Comment (guperrot) [inline SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Windows.Shared/Crashes.cs]:
```suggestion
        private static void PlatformUnsetInstance()
```

File: SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.iOS/Crashes.cs [hunk 3]
```diff
@@ -287,5 +287,10 @@ private static NSDictionary StringDictToNSDict(IDictionary<string, string> dict)
         {
             return NSDictionary.FromObjectsAndKeys(dict.Values.ToArray(), dict.Keys.ToArray());
         }
+
+        internal static void PlatformUnsetInstance()
```
Comment (guperrot) [inline SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.iOS/Crashes.cs]:
```suggestion
        private static void PlatformUnsetInstance()
```

The platform specific method is called only by shared code of the same partial class. Other classes must call the shared UnsetInstance that is internal.

File: SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes/Crashes.cs [hunk 4]
```diff
@@ -43,5 +43,9 @@ private static void PlatformNotifyUserConfirmation(UserConfirmation confirmation
         private static void PlatformTrackError(Exception exception, IDictionary<string, string> properties, ErrorAttachmentLog[] attachments)
         {
         }
+
+        internal static void PlatformUnsetInstance()
```
Comment (guperrot) [inline SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes/Crashes.cs]:
```suggestion
        private static void PlatformUnsetInstance()
```

File: Tests/Contoso.Test.Functional.Droid/MainActivity.cs [hunk 5]
```diff
@@ -3,21 +3,30 @@
...
 
 namespace Contoso.Test.Functional.Droid
 {
+    using Distribute = Microsoft.AppCenter.Distribute.Distribute;
+
     [Activity(Label = "xUnit Android Runner", MainLauncher = true, Theme = "@android:style/Theme.Material.Light")]
     public class MainActivity : RunnerActivity
     {
-        private const string ResultChannelHost = "10.0.2.2";
+        public static string ResultChannelHost = "10.0.2.2";
```
Comment (guperrot) [inline Tests/Contoso.Test.Functional.Droid/MainActivity.cs]: What is this change?

File: Tests/Microsoft.AppCenter.Test.Functional/HttpNetworkAdapter.cs [hunk 6]
```diff
@@ -1,62 +1,71 @@
...
-using Newtonsoft.Json.Linq;
 
 namespace Microsoft.AppCenter.Test.Functional
 {
     internal class HttpNetworkAdapter : IHttpNetworkAdapter
     {
-        private readonly int _expectedStatusCode;
-        private readonly string _expectedContent;
-        private readonly string _expectedLogType;
+        struct ExpectedData
```
Comment (guperrot) [inline Tests/Microsoft.AppCenter.Test.Functional/HttpNetworkAdapter.cs]: It's more conventional for us to use a class and properties.

**Pull request merged**